core/src/main/scala/com/nvidia/spark/rapids/tool/qualification/Qualification.scala

```diff
@@ -114,9 +117,33 @@ class Qualification(outputDir: String, numRows: Int, hadoopConf: Configuration,
   private def sortForExecutiveSummary(appsSumDesc: Seq[QualificationSummaryInfo],
       order: String): Seq[EstimatedSummaryInfo] = {
     if (QualificationArgs.isOrderAsc(order)) {
-      appsSumDesc.reverse.map(_.estimatedInfo)
+      appsSumDesc.reverse.map(sum =>
+        EstimatedSummaryInfo(sum.estimatedInfo, sum.estimatedFrequency))
     } else {
-      appsSumDesc.map(_.estimatedInfo)
+      appsSumDesc.map(sum => EstimatedSummaryInfo(sum.estimatedInfo, sum.estimatedFrequency))
+    }
+  }
+
+  // Estimate app frequency based off of all applications in this run, unit jobs per month
+  private def estimateAppFrequency(
+    appsSum: Seq[QualificationSummaryInfo]): Seq[QualificationSummaryInfo] = {
+    val appFrequency = scala.collection.mutable.Map[String, Double]()
+    var windowStart: Long = Long.MaxValue
+    var windowEnd: Long = Long.MinValue
+
+    appsSum.foreach { sum =>
+      appFrequency += (sum.appName -> (1.0 + appFrequency.getOrElse(sum.appName, 0.0)))
+      windowStart = Math.min(sum.startTime, windowStart)
+      windowEnd = Math.max(windowEnd, sum.startTime + sum.estimatedInfo.appDur)
+    }
+    val windowInMonths =
+      if (windowEnd > windowStart) ((windowEnd - windowStart) / (1000.0*60*60*24*30)) else 1.0
```

Comment:
I suggest that internal representation of the frequency to be "daily" based. Then the final output can be switched to 30 if necessary so that we can configure that easily.
Also, dividing by 30.0 risks that we get lots of Zeros by making the denominator much bigger than the numerator.